lib/bridge/PublicitySyncer.js

```diff
@@ -181,8 +181,12 @@ PublicitySyncer.prototype._solveVisibility = Promise.coroutine(function*() {
         let currentState = this._visibilityMap.roomVisibilities[roomId];
         let correctState = shouldBePrivate(roomId, []) ? 'private' : 'public';
 
+        // Use the server domain of the first mapping
+        // 'localhost #channel1' => 'localhost'
+        const networkId = this._visibilityMap.mappings[roomId][0].split(' ')[0];
```

Comment:
The null guard shouldn't be needed, you're right. And yes the mappings are guaranteed to be set for a room, and contain at least one channel mapping otherwise that room wouldn't be in the `RoomStore`